plugins/kubesvc/deployment.yaml

```diff
@@ -0,0 +1,76 @@
+apiVersion: apps/v1
```

Comment:
Can we instead use a manifest at https://armory.jfrog.io/artifactory/manifests/kubesvc/?
e.g. 
```
...
bases:
  - https://armory.jfrog.io/artifactory/manifests/kubesvc/kubesvc-0.2.0-rc.44-kustomize.tar.gz  

configMapGenerator:
- name: kubesvc-config
  behavior: merge
  files:
  - patches/kubesvc.yaml
```

(that's what I use for testing kubesvc). We could add a new base kustomization to kubesvc and let you override the image here as well.

Reply:
There are a couple of reasons for having the manifests here:
* After talking with SAs they see it valuable to know exactly what is being deployed, and referencing remote files doesn't work in air gapped environments.
* By having the manifests here we're not blocking users if they need to make any change to adjust for their environment. For example this deployment manifest is already mounting the `spin-secrets` volume that is used for the rest of the secrets in this repo, so deploying kubesvc works out of the box.

I know there's duplication but that comes at the cost of flexibility. It's the same reason for example, of having to add `spin-gate-expose.yml` to this repo, even though there are already ways for generating and tweaking gate service manifest, there are use cases where users just want to have full control over the manifests.